pkg/orchestration/wiring/ec2compute/common/common.go

```diff
@@ -115,46 +115,60 @@ func WireUp(microServiceNameInSpec, ec2ComputePlanName string, stateResource *or
 	}
 
 	var bindingResources []smith_v1.Resource
-	var bindingResult []compute.BindingResult
+	var resourcesWithEnvVarBindings []compute.ResourceWithEnvVarBinding
+	var resourcesWithIamAccessibleBindings []iam.ResourceWithIamAccessibleBinding
 	var references []smith_v1.Reference
 
 	for _, dependency := range dependencies {
-		bindableShape, found, err := knownshapes.FindBindableEnvironmentVariablesShape(dependency.Contract.Shapes)
+		bindableEnvVarShape, envVarFound, err := knownshapes.FindBindableEnvironmentVariablesShape(dependency.Contract.Shapes)
 		if err != nil {
 			return nil, false, err
 		}
-		if !found {
+		if !envVarFound {
 			return nil, false, errors.Errorf("cannot depend on resource %q, only dependencies providing shape %q are supported", dependency.Name, knownshapes.BindableEnvironmentVariablesShape)
 		}
 
-		resourceReference := bindableShape.Data.ServiceInstanceName
+		resourceReference := bindableEnvVarShape.Data.ServiceInstanceName
 		bindingResource := wiringutil.ConsumerProducerServiceBinding(stateResource.Name, dependency.Name, resourceReference)
 		bindingResources = append(bindingResources, bindingResource)
-		bindingResult = append(bindingResult, compute.BindingResult{
-			ResourceName:            dependency.Name,
-			BindableEnvVarShape:     *bindableShape,
-			CreatedBindingFromShape: bindingResource,
+		resourcesWithEnvVarBindings = append(resourcesWithEnvVarBindings, compute.ResourceWithEnvVarBinding{
+			ResourceName:        dependency.Name,
+			BindableEnvVarShape: *bindableEnvVarShape,
+			BindingName:         bindingResource.Name,
 		})
-	}
 
-	// filter out the list of dependencyReferences - this will be used
-	dependencyReferences := make([]smith_v1.Reference, 0, len(bindingResult))
-	for _, res := range bindingResult {
-		dependencyReferences = append(dependencyReferences, smith_v1.Reference{
-			Resource: res.CreatedBindingFromShape.Name,
-		})
+		// We also depend on BindableIamAccessible shape
+		bindableIamAccessibleShape, iamFound, err := knownshapes.FindBindableIamAccessibleShape(dependency.Contract.Shapes)
```

Comment:
What if there are two?

Reply:
Two shapes with the same name returned by an autowiring functions is an error. https://github.com/atlassian/voyager/pull/78

Reply:
From a single wiring function, sure. But what stops that from happening from two different wiring plugins?